### PR TITLE
Rewired deps to public releases, fixed up a few more things

### PR DESCRIPTION
**Pull request by mikael-s-persson**
Rewired deps to public releases, fixed up a few more things

Added codeql config for checking.
Rewired dependencies to http archives of public releases.
Fixed the sha and version numbers of absl, protobuf and eigen.
Removed the submodules.

### CODE REVIEW

**Review by bingjeff — APPROVED** [review]

**Comment by mikael-s-persson** [comment]
@bingjeff Thanks for the approval! Can you hit the merge button? Thanks.

**Comment by mmaraya** [comment]
Approving to unblock code scanning.

**Pull request merged**
